app/api/submission/Consignment.scala

```diff
@@ -58,20 +61,65 @@ object locationOfGoodsType01 {
     case _                    => throw new Exception("Invalid type of location value")
   }
 
-  implicit val reads: Reads[LocationOfGoodsType01] = (
-    (__ \ "typeOfLocation").read[String].map(convertTypeOfLocation) and
-      (__ \ "qualifierOfIdentification").read[String].map(convertQualifierOfIdentification) and
+  def chooseLocationOfGoodsReads(isSimplified: Boolean): Reads[LocationOfGoodsType01] =
```

Comment:
Interesting. This is actually really good work. However I'm of the opinion that these reads should be as "dumb" as possible. We want to put faith in our frontend navigation. So, for example, if "typeOfLocation" is not populated in user answers, then we know it's a simplified journey and that it's "B". We can express this in the reads with
`(__ \ "typeOfLocation").readWithDefault[String]("authorisedPlace").map(convertTypeOfLocation)`. Same with the qualifier of identification `(__ \ "qualifierOfIdentification").readWithDefault[String]("authorisationNumber").map(convertQualifierOfIdentification)`

Reply:
Could then store these strings as vals in the locationOfGoodsType01 object as they're being used twice